src/components/button/button_content.tsx

```diff
@@ -109,7 +109,9 @@ export const OuiButtonContent: FunctionComponent<
 
   return (
     <span {...contentProps} className={contentClassNames}>
-      {buttonIcon}
+      <span {...textProps} className="">
```

Comment:
```suggestion
      <span {...textProps}>
```

Reply:
Hi @BSFishy,

Thanks for the comment. Regarding your suggestion of removing `className=""` I already explained @joshuarrrr and @kgcreative the reason behind this in [this](https://github.com/opensearch-project/oui/issues/1129#issuecomment-1804846572) comment. Here that explanation again below:

>The dictionary textProps for the case of iconOnly button contains the following:

>{className: 'ouiButton__text ouiButtonGroupButton__textShift', data-text: 'Option three', title: 'Option three', ref: ƒ}

>By adding, className="" after {...textProps} we override the initial definition of className coming in textProps. This simple solution works allowing us to visualize the tooltip. Of course, I feel it is not quite elegant, and maybe we should move the HTML attributes data-text and title to the contentProps

I can create some custom styling to create a more ad-hoc solution if you want

Reply:
Okay I see. In that case, I think it would be better to copy `textProps` and extract `className` out of it. Adding `className` like this is a clever solution, but it isn't clear just by code what it's doing. Additionally, it muddies up the snapshots with empty classnames.

Maybe something like

```tsx
const iconTextProps = { ...textProps };
iconTextProps.className = undefined;
```

Reply:
Yeah! I new my solution was not elegant. Thanks for the suggestion @BSFishy. I will implement it and let you know when I the update is ready

Reply:
Thinking about it more, there are potentially a few other options

Something like

```tsx
const { className: _, ...iconTextProps } = textProps;
```

or doing

```tsx
<span {...textProps} className={undefined}>
```

I think should also work. The only reason I bring it up is because my first suggestion might unnecessarily create memory, but I don't know do whatever sounds good to you

Reply:
I went with the second option. Code updated. Let me know if you have any concerns.